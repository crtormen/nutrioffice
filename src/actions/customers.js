import database from '../firebase/firebase';

//Action generator ADD_CUSTOMER
export const addCustomer = (customer) => ({
    type: 'ADD_CUSTOMER',
    customer
});

//Pass a function instead of a object  
// export const startAddCustomer = (customerData = {}) => {
//     return (dispatch, getState) => { 
//         const uid = getState().auth.uid;
//         const { //default values
//             name = '', 
//             email = '',
//             phone = '', 
//             born = '',              
//             occupation = '', 
//             instagram = ''
//         } = customerData;
//         const customer = { name, email, phone, born, occupation, instagram };

//         return database.ref(`users/${uid}/customers`).push(customer).then((ref) => { 
//             dispatch(addCustomer({
//                 id: ref.key,
//                 ...customer
//             }));
//         });
//     };
// };

//Action generator for EDIT_CUSTOMER
export const editCustomer = (id, updates) => ({
    type: 'EDIT_CUSTOMER',
    id,
    updates
});

// export const startEditExpense = (id, updates) => {
//     return (dispatch, getState) => {
//         const uid = getState().auth.uid;
//         return database.ref(`users/${uid}/expenses/${id}`).update(updates).then(() => {
//             dispatch(editExpense(id, updates));
//         })
//     };
// };

//Action generator for REMOVE_CUSTOMER
export const removeCustomer = ({ id } = {}) => ({
    type: 'REMOVE_CUSTOMER',
    id
});

// export const startRemoveExpense = ({ id } = {}) => {
//     return (dispatch, getState) => {
//         const uid = getState().auth.uid;
//         return database.ref(`users/${uid}/expenses/${id}`).remove().then(() => {
//             dispatch(removeExpense({ id }));
//         });
//     };
// };


//Action generator for SET_CUSTOMERS
export const setCustomers = (customers) => ({
    type: 'SET_CUSTOMERS',
    customers
});

//Action Generator START_SET_EXPENSES
// export const startSetExpenses = () => {
//     return (dispatch, getState) => {
//         const uid = getState().auth.uid;
//         //returns a promises, so that when using startSetExpenses, we can use "then"
//         return database.ref(`users/${uid}/expenses`).once('value').then((snapshot) => {
//             let expenses = [];
//             snapshot.forEach((childSnapshot) => {
//                 expenses.push({
//                     id: childSnapshot.key,
//                     ...childSnapshot.val()
//                 })
//             });

//             dispatch(setExpenses(expenses));
//         });        
//     };
// };
