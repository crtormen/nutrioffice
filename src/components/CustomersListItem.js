import React from 'react';
import { Link } from 'react-router-dom';

const CustomersListItem = ({id, name, email}) => (
    <div>
        <h3><Link to={`edit/${id}`}>{name}</Link> - {email}</h3>    
    </div>
);

export default CustomersListItem;