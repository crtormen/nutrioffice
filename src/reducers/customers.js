const customerReducerDefaultState = [];

export default (state = customerReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_CUSTOMER':
            return [...state, action.customer];
        case 'EDIT_CUSTOMER':
            return state.map((customer) => {
                if (customer.id === action.id) {
                    return {
                        ...customer,
                        ...action.updates //create new object based ond the previous, with updates and concatenations
                    }
                }
                return customer; //do nothing if not found
            });
        case 'REMOVE_CUSTOMER':
            return state.filter(({ id }) => id !== action.id ); ////return all in a new array, except that one where filter not match 
        case 'SET_CUSTOMERS':
            return action.customers;
        default: 
            return state;
    };
};