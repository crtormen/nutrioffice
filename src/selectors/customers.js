export default (customers) => {
    return customers;
}



// Get visible expenses, filtering by text, period of time, and allows to sort by date and amount
// export default (expenses, { text, sortBy, startDate, endDate }) => { 
//     return expenses.filter((expense) => {
//         const startDateMatch = startDate ? startDate.isSameOrBefore(moment(expense.createdAt), 'day') : true;
//         const endDateMatch = endDate ? endDate.isSameOrAfter(moment(expense.createdAt), 'day') : true;
//         const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());

//         return startDateMatch && endDateMatch && textMatch;
//     }).sort((a, b) => {  //highest first
//         if (sortBy === 'date')
//             return a.createdAt < b.createdAt ? 1 : -1;
//         if (sortBy === 'amount')
//             return a.amount < b.amount ? 1 : -1;

//     });
// };
