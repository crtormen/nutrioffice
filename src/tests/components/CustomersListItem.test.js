import React from 'react';
import { shallow } from 'enzyme';
import CustomersListItem from '../../components/CustomersListItem';
import { customers } from '../fixtures/customers';


test("should render CustomersListItem component", () => {
    const wrapper = shallow(<CustomersListItem {...customers[0]} />);
    expect(wrapper).toMatchSnapshot();
});
