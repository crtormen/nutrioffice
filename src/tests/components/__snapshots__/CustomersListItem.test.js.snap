// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CustomersListItem component 1`] = `
<div>
  <h3>
    <Link
      replace={false}
      to="edit/1"
    >
      Claudio
    </Link>
     - 
    crtormen@gmail.com
  </h3>
</div>
`;
